Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/debuglog/ui/DebugLogFragment.kt

```diff
@@ -94,41 +99,48 @@ class DebugLogFragment : Fragment(R.layout.bugreporting_debuglog_fragment), Auto
             }
         }
 
-        vm.shareEvent.observe2(this@DebugLogFragment) {
-            startActivityForResult(it.createIntent(), it.id)
-        }
-
-        vm.logStoreResult.observe2(this) {
-            Toast.makeText(
-                requireContext(),
-                "TODO: Show store result dialog: ${it.storageUri}",
-                Toast.LENGTH_LONG
-            ).show()
+        vm.events.observe2(this) {
+            when (it) {
+                DebugLogViewModel.Event.ShowLogDeletedConfirmation -> {
+                    showLogDeletionConfirmation()
+                }
+                DebugLogViewModel.Event.NavigateToPrivacyFragment -> {
+                    doNavigate(
+                        DebugLogFragmentDirections.actionDebuglogFragmentToInformationPrivacyFragment()
+                    )
+                }
+                DebugLogViewModel.Event.NavigateToShareFragment -> {
+                    doNavigate(
+                        DebugLogFragmentDirections.actionDebuglogFragmentToLogUploadHistoryFragment()
+                    )
+                }
+                DebugLogViewModel.Event.NavigateToUploadHistory -> {
+                    doNavigate(
+                        DebugLogFragmentDirections.actionDebuglogFragmentToDebugLogUploadFragment()
+                    )
+                }
+                is DebugLogViewModel.Event.Error -> {
+                    Toast.makeText(requireContext(), it.toString(), Toast.LENGTH_LONG).show()
+                }
+                DebugLogViewModel.Event.ShowLowStorageDialog -> {
+                    showLowStorageError()
+                }
+                is DebugLogViewModel.Event.LocalExport -> {
+                    startActivityForResult(it.request.createIntent(), it.request.id)
+                }
+                is DebugLogViewModel.Event.ExportResult -> {
+                    showExportResult()
+                }
+                is DebugLogViewModel.Event.ShowLocalExportError -> {
+                    showLocalExportError(it.error)
+                }
```

Comment:
can you please remove `{}` ? it is one statement for every case , I think it will be more readable

Reply:
I tried this, but not all elements within this  `when` can be written in a single line statement, so some wrap and some don't, when applying it for every element, I find the whole `when` was "shorter" but imho less readable. If this isn't a blocker for you, I'd keep it as is.

Reply:
ok